tests/transformers/test_transformers_prompt_templating.py

```diff
@@ -0,0 +1,171 @@
+    "automatic-speech-recognition",
+    "text-to-audio",
+    "text-to-speech",
+    "text-classification",
+    "sentiment-analysis",
+    "token-classification",
+    "ner",
+    "question-answering",
+    "table-question-answering",
+    "visual-question-answering",
+    "vqa",
+    "document-question-answering",
+    "translation",
+    "zero-shot-classification",
+    "zero-shot-image-classification",
+    "zero-shot-audio-classification",
+    "conversational",
+    "image-classification",
+    "image-segmentation",
+    "image-to-text",
+    "object-detection",
+    "zero-shot-object-detection",
+    "depth-estimation",
+    "video-classification",
+    "mask-generation",
+    "image-to-image",
+]
+
+
+@pytest.fixture(scope="session")
+def small_text_generation_model():
+    return pipeline("text-generation", model="distilgpt2")
+
+
+@pytest.fixture(scope="session")
+def saved_transformers_model_path(tmp_path_factory, small_text_generation_model):
+    tmp_path = tmp_path_factory.mktemp("model")
+    mlflow.transformers.save_model(
+        transformers_model=small_text_generation_model,
+        path=tmp_path,
+        prompt_template=TEST_PROMPT_TEMPLATE,
+    )
+    return tmp_path
+
+
+@pytest.mark.parametrize(
+    "template",
+    [
+        "{multiple} {placeholders}",
+        "No placeholders",
+        "Placeholder {that} isn't `prompt`",
+        "Placeholder without a {} name",
+        "Placeholder with {prompt} and {} empty",
+        1001,  # not a string
+    ],
+)
+def test_prompt_validation_throws_on_invalid_templates(template):
+    match = (
+        "Argument `prompt_template` should be a string with a single format arg, 'prompt'."
+        if isinstance(template, str)
+        else "Argument `prompt_template` should be a string"
+    )
+    with pytest.raises(MlflowException, match=match):
+        _validate_prompt_template(template)
+
+
+@pytest.mark.parametrize(
+    "template",
+    [
+        "Single placeholder {prompt}",
+        "Text can be before {prompt} and after",
+        # the formatter will interpret the double braces as a literal single brace
+        "Escaped braces {{ work fine {prompt} }}",
+    ],
+)
+def test_prompt_validation_succeeds_on_valid_templates(template):
+    assert _validate_prompt_template(template) is None
+
+
+# test that prompt is saved to mlmodel file and is present in model load
+def test_prompt_save_and_load(saved_transformers_model_path):
+    mlmodel_path = saved_transformers_model_path / MLMODEL_FILE_NAME
+    with open(mlmodel_path) as f:
+        mlmodel_dict = yaml.safe_load(f)
+
+    assert mlmodel_dict["prompt_template"] == TEST_PROMPT_TEMPLATE
+
+    model = mlflow.pyfunc.load_model(saved_transformers_model_path)
+    assert model._model_impl.prompt_template == TEST_PROMPT_TEMPLATE
+
+
+def test_saving_prompt_throws_on_unsupported_task():
+    model = pipeline("text-generation", model="distilgpt2")
+
+    for pipeline_type in UNSUPPORTED_PIPELINES:
```

Comment:
Can we juts choose one and test it?

Reply:
haha yeah i figured this was a little unnecessary, but it seemed pretty cheap since the save_model gets short-circuited by the throw anyway